### PR TITLE
Path utils refactor

### PR DESCRIPTION
**Pull request by RussKie**
As discussed in https://github.com/gitextensions/gitextensions/pull/4032#issuecomment-333517989

* Deprecate PathUtil.DirectoryExists
* Deprecate PathUtil.PathExists
* Move PathEqualityComparer to outer scope

### CODE REVIEW

File: GitCommands/PathUtil.cs [hunk 1]
```diff
@@ -188,36 +156,9 @@ public static bool IsValidPath(string aPath)
             return fi != null;
         }
                 
-        public static bool PathExists(string aPath)
```
Comment (jbialobr) [inline GitCommands/PathUtil.cs]: These methods was created to return false for cases when the given path is of incorrect form (causing exceptions being caught - lines 198-200). The uses case is for expanding to a full path based on environment PATH variable. We are seeking only for valid paths and don't want to propagate exceptions to UI.

**Comment by RussKie** [comment]
Thanks, I'll make sure exceptions here are contained

**Comment by RussKie** [comment]
`File.Exists` doesn't throw, but I added few unit tests just in case

**Comment by jbialobr** [comment]
The changes are not semantically equal. The current implementation stops the seeking process after the first exception being thrown. The previous implementation kept seeking after an exception being thrown.

**Comment by RussKie** [comment]
That is actually not true - they are equal.
I've added unit tests before I've wrapped the implementations in try/catch. 

There are two reasons why I've added try/catch:
1. passing `null` causes `File.Exists` to throw, and it would throw in the original implementation as well
2. `TryXxx` methods should not throw in any case

`Environment` needs to be abstracted so `Environment.Get/SetEnvironmentVariable` can be tested, then more tests can be written to test this functionality

**Comment by jbialobr** [comment]
```csharp
                foreach (var path in GetEnvironmentValidPaths())
                {
                    fullPath = Path.Combine(path, fileName);
the created fullPath can be too long giving PathTooLongException
                    if (File.Exists(fullPath))
                    {
                        return true;
                    }
                }
```

**Comment by RussKie** [comment]
image

**Comment by RussKie** [comment]
That works on both W8 and W10

**Comment by jbialobr** [comment]
Yes, you are right. Now I see that the root cause for #2924 was `Path.Combine`, not `Path.Exists` :)

**Pull request merged**